neo/core/container.py

```diff
@@ -74,23 +77,27 @@ def filterdata(data, targdict=None, objects=None, **kwargs):
         results = []
         for key, value in sorted(targdict.items()):
             for obj in data:
-                if (hasattr(obj, key) and getattr(obj, key) == value and
-                        all([obj is not res for res in results])):
+                if (hasattr(obj, key) and getattr(obj, key) == value):
+                    results.append(obj)
+                if (isinstance(value, filters.FilterCondition)) and (
+                        key in obj.annotations and value.evaluate(obj.annotations[key])):
                     results.append(obj)
-                elif (key in obj.annotations and obj.annotations[key] == value and
-                          all([obj is not res for res in results])):
+                if key in obj.annotations and obj.annotations[key] == value:
                     results.append(obj)
 
+    # remove duplicates from results
+    res = []
+    [res.append(result) for result in results if not any([result is elem for elem in res])]
```

Comment:
This looks as though it will be slow - perhaps use a `set` to eliminate duplicates?

Reply:
perhaps use the Python `id()` function?, e.g., 
```
unique_results = { id(res): res for res in results }.values()
```

Reply:
Perfect, thanks!